voicebox/encryption.py

```diff
@@ -0,0 +1,240 @@
+import logging
+
+import secrets
+
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.ciphers import (
+    Cipher,
+    algorithms,
+    modes
+)
+from cryptography.hazmat.backends import default_backend
+
+
+class BaseEncryptor:
+
+    KEY_EXCHANGE_SIGNAL = 900
+
+    def __init__(self, *args, **kwargs):
+
+        raise NotImplementedError("Cannot Initialize from BaseEncryptor")
+
+    def encrypt(self, payload: bytes):
+
+        raise NotImplementedError()
+
+    def decrypt(self, packet: bytes):
+
+        raise NotImplementedError()
+
+    def hash(self, payload):
+
+        raise NotImplementedError()
+
+    def packet_handler(self, packet):
+
+        return packet
+
+    @classmethod
+    @property
+    def public_pem(cls):
+
+        return None
+
+
+class RSAEncryptor(BaseEncryptor):
+
+    KEY_SIZE = 2048
+
+    __private_key = None
+
+    __public_key = None
+
+    INT_BYTE_SIZE = 4
+
+    KEY_EXCHANGE_SIGNAL = 901
```

Comment:
It's a good practice to avoid using magic numbers. Consider replacing the `901` with a constant that describes what the number represents. [important]